### DISCUSSION TITLE
Approach to reusing query results for other queries

### DISCUSSION THREAD

**Discussion by skokenes** [duckdb/duckdb#6760] [Q&A]
My question is about how to create cached query results to feed other queries.

Imagine a dashboard UI where a user has defined a filter condition that should affect all charts on the page. All charts are queries from the same table. The filter condition should be applied to that table before the charts, then execute their queries. Assume the table is large and the filter condition is slow, so I don't want to repeat it for every chart query.

What is the best way to cache these intermediate results for re-use downstream? One thought is to use a temporary table via `CREATE OR REPLACE TEMP TABLE` every time the filter condition changes, and then use that table for subsequent queries.

Any alternative ways to query a predetermined set of filtered rows repeatedly?

**Comment by hannes** [comment]
Indeed creating a temp table is probably your best bet here.
